src/v/cloud_storage/segment_state.cc

```diff
@@ -18,6 +18,10 @@
 namespace cloud_storage {
 
 void materialized_segment_state::offload(remote_partition* partition) {
+    if (parent && parent->_gate.is_closed()) {
```

Comment:
This shouldn't be necessary once we fix stop() to unlink from the embedded list
https://github.com/redpanda-data/redpanda/pull/7832#discussion_r1052320336

Reply:
I forgot to remove this, will fix